File: Algorithm/Program Class/Java/Class2/src/com/two/Test05.java
package com.two;

public class Test05 {
	public static void main(String[] args) {

		int intput = 2991;

		while (true) {
			intput++;
			if (intput / 10000 > 0) {
				break;
			} else {
				int sum = intput / 1000 + intput / 100 % 10 + intput % 100 / 10 + intput % 100 % 10;
				int temp;
				int sum_2 = 0;
				temp = intput;
				while (temp > 0) {
					sum_2 += temp % 16;
					temp = temp / 16;
				}
				int sum_3 = 0;
				temp = intput;
				while (temp > 0) {
					sum_3 += temp % 12;
					temp = temp / 12;
				}
				if (sum == sum_2 && sum_2 == sum_3)
					System.out.println(intput);
				else
					continue;
			}
		}

	}

}

File: Algorithm/Class/C C++ Python/Class 2/Test05/Test05.cpp
#include <iostream>
using namespace std;

int main()
{
    int intput = 2991;

    while(intput/10000==0)
    {
        intput++;
        int sum = intput/1000 + intput/100%10 + intput%100/10 +intput%10;
        int value = intput;
        int sum_o = 0;

        while(value)
        {
            sum_o += value % 12;

            value = value /12;
        }
       // cout<<"12 "<<sum_o<<" "<<sum<<endl;
        if(sum_o != sum)
            continue;
        sum_o = 0;
        value = intput;
        while(value)
        {
            sum_o += value % 16;
            value = value /16;
        }
        if(sum!=sum_o)
            continue;
        cout<<"16 "<<sum_o<<" "<<sum<<endl;

        cout<<intput<<endl;

    }
    return 0 ;
}
